src/main/java/seedu/exercise/storage/serializablebook/JsonSerializableScheduleBook.java

```diff
@@ -0,0 +1,37 @@
+package seedu.exercise.storage.serializablebook;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonRootName;
+
+import seedu.exercise.model.book.ReadOnlyResourceBook;
+import seedu.exercise.model.schedule.Schedule;
+import seedu.exercise.storage.resource.JsonAdaptedSchedule;
+
+/**
+ * An Immutable ScheduleBook that is serializable to JSON format.
+ */
+@JsonRootName(value = "schedulebook")
+public class JsonSerializableScheduleBook extends SerializableResourceBook<JsonAdaptedSchedule> {
+
+    @JsonCreator
+    public JsonSerializableScheduleBook(@JsonProperty("jsonResources") List<JsonAdaptedSchedule> schedules) {
+        super(schedules);
+    }
+
+    public JsonSerializableScheduleBook(ReadOnlyResourceBook<Schedule> source) {
+        List<JsonAdaptedSchedule> schedules = new ArrayList<>();
+        schedules
+            .addAll(source.getResourceList()
+                .stream()
+                .map(JsonAdaptedSchedule::new)
+                .collect(Collectors.toList()));
+        jsonResources.addAll(schedules);
+    }
```

Comment:
Is it possible to abstract all this method out into the superclass `SerializableResourceBook` or a Util class for Storage? Because this method look very similar to the other constructors for `JsonSerializableRegimeBook` and `JsonSerializableExerciseBook`.

Reply:
Have abstracted this method out into ``SerializableResourceBook``. 

One way which people suggest to avoid the unchecked cast warning is to pass the type parameter together in the generic function. Please do feedback if any of you have issues with that!

Reply:
That method is fine for me.